### PR TITLE
Add healthchecks to docker-compose.yml

### PR DESCRIPTION
**Pull request by ftoppi**
Hello,

here's a small PR to replace redis with keydb due to redis license changes and to add healthchecks to services.

### CODE REVIEW

File: docker-compose.yml
Comment (ostefano) [inline docker-compose.yml]: This image is x64 only while the Redis image is ARM as well. Would it be possible to remove the arch modifier here?

File: docker-compose.yml
Comment (ostefano) [inline docker-compose.yml:66]: Is this check run inside the container? In other words, Is `localhost` the correct setting regardless of our `BASE_URL`?

File: docker-compose.yml
Comment (ostefano) [reply docker-compose.yml]: Note that unless MISP changes the default key-value storage, we should only document this alternative in the README.md rather than modify the `docker-compose.yml` file

Comment (ftoppi) [reply docker-compose.yml]: Fine by me. All their images are arch-specific except `eqalpha/keydb:latest` which is both for arm64 and x86_64.

File: docker-compose.yml
Comment (ftoppi) [reply docker-compose.yml:66]: Yes, it's run inside the container.
You can see below I don't expose ports in my stack and it's still healthy:

```
$ docker compose ps 2> /dev/null
NAME                  IMAGE                                          COMMAND                  SERVICE        CREATED        STATUS                  PORTS
misp-db-1             mariadb:10.11                                  "docker-entrypoint.s…"   db             43 hours ago   Up 43 hours (healthy)   3306/tcp
misp-mail-1           ixdotai/smtp                                   "/bin/entrypoint.sh …"   mail           43 hours ago   Up 43 hours             25/tcp
misp-misp-core-1      ghcr.io/misp/misp-docker/misp-core:latest      "/entrypoint.sh"         misp-core      42 hours ago   Up 42 hours (healthy)
misp-misp-modules-1   ghcr.io/misp/misp-docker/misp-modules:latest   "/usr/local/bin/misp…"   misp-modules   43 hours ago   Up 43 hours
misp-redis-1          eqalpha/keydb:x86_64_v6.3.4                    "docker-entrypoint.s…"   redis          43 hours ago   Up 43 hours (healthy)   6379/tcp
```

File: docker-compose.yml
Comment (ostefano) [reply docker-compose.yml]: We should also reference https://github.com/valkey-io/valkey if a docker image is available.

File: docker-compose.yml
Comment (ostefano) [inline docker-compose.yml]: What is the alternative command if were to use Redis here?

**Review by ostefano — CHANGES_REQUESTED** [review]
This is good stuff, but I think it's too early to replace redis.

Can we add the healthchecks first, and only when better docker images are published, move to a new key value store?

Alternatively, it should be easy to just document the alternatives available in the README.

File: docker-compose.yml
Comment (ftoppi) [reply docker-compose.yml]: redis-cli ping

**Review by ostefano — CHANGES_REQUESTED** [review]
Will merge if changes to the underlying image are removed.

Also, note that we decided to migrate to valkey, so healthcheck commands might need updating.

**Comment by ftoppi** [comment]
> Will merge if changes to the underlying image are removed.
> 
> Also, note that we decided to migrate to valkey, so healthcheck commands might need updating.

I figured everyone would move to valkey but it wasn't available at the time.

This is better?

**Review by ostefano — APPROVED** [review]

**Comment by ostefano** [comment]
Looks good.

Can you rebase and squash? Otherwise I will do it when merging (will wait for next MISP release).

**Pull request merged**
